spring-cloud-aws-messaging/src/main/java/io/awspring/cloud/messaging/endpoint/config/NotificationHandlerMethodArgumentResolverFactoryBean.java

```diff
@@ -34,18 +36,17 @@
 
 	private final AmazonSNS amazonSns;
 
-	private final SnsMessageManager snsMessageManager;
+	private final Optional<SnsMessageManager> snsMessageManager;
```

Comment:
Ideally we don't have `Optional` fields. Instead this field can be just nullable like everywhere else in the code. Add a Javadoc that it can be null to make it clear.